cli/napi/function.rs

```diff
@@ -0,0 +1,161 @@
+use deno_core::napi::*;
+
+#[repr(C)]
+#[derive(Debug)]
+pub struct CallbackInfo {
+  pub env: napi_env,
+  pub cb: napi_callback,
+  pub cb_info: napi_callback_info,
+  pub args: *const c_void,
+}
+
+pub unsafe fn create_function<'a>(
+  env: &'a mut Env,
+  name: Option<&str>,
+  cb: napi_callback,
+  cb_info: napi_callback_info,
+) -> v8::Local<'a, v8::Function> {
+  let method_ptr = v8::External::new(env.scope, std::mem::transmute(cb));
```

Comment:
Would there be a "safer" alternative available for transmute? Not that I particularly doubt the correctness, it's just a bit odd to see it here. PR-reading wise, it would be easier if the transmute types were explicitly written out although they're unnecessary from a code perspective.

Reply:
AFAIK the only was to cast to function pointers is using `transmute`. I agree on making the transmuted types explicit 👍

Reply:
Actually even clippy suggested this, so I've changed it to cast now

![image](https://user-images.githubusercontent.com/43033058/153228086-0da2d0ab-f30c-481a-9675-32eafc4a8a3b.png)

Reply:
Casting from function to pointer is fine but it is not possible the other way around (a few lines below) IIRC

Reply:
Oh, yes, still using transmute for that